fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/observer/FedoraEventImpl.java

```diff
@@ -255,5 +253,15 @@ public static FedoraEvent from(final Event event) {
         }
     }
 
+    private static String cleanPath(final Event event) throws RepositoryException {
+        // remove any trailing data for property changes
+        final String path = PROPERTY_TYPES.contains(event.getType()) ?
+            event.getPath().substring(0, event.getPath().lastIndexOf("/")) : event.getPath();
+
+        // reformat any hash URIs and remove any trailing /jcr:content
+        final HashConverter converter = new HashConverter();
```

Comment:
Sadly, no, the `Converter` classes are all stateful. This is another reason why it may be worthwhile to refactor our use of these away from Guava's `Converter` classes, but that is WAAAAY beyond the scope of this ticket. (They could easily be refactored into pairs of curried functions, but it would have wide-ranging implications for the entire codebase.)